src/test/java/org/broadinstitute/hellbender/utils/pairhmm/VectorPairHMMUnitTest.java

```diff
@@ -23,22 +24,27 @@
 
     private static final String pairHMMTestData = publicTestDir + "pairhmm-testdata.txt";
 
-   @BeforeClass
-    public void initialize() {
-       try {
-           new VectorLoglessPairHMM(null);
-       } catch (final Exception e) {
-           throw new SkipException("AVX PairHMM is not supported on this system or the library is not available");
-       }
-    }
-
+    // Return a list of supported VectorLoglessPairHMM implementations, skip the test if none are supported
     private List<N2MemoryPairHMM> getHMMs() {
-        final PairHMMNativeArguments args = new PairHMMNativeArguments();
+        List<N2MemoryPairHMM> list = new ArrayList<>();
+        PairHMMNativeArguments args = new PairHMMNativeArguments();
         args.useDoublePrecision = false;
         args.maxNumberOfThreads = 1;
-        final N2MemoryPairHMM avxPairHMM = new VectorLoglessPairHMM(args);
-        avxPairHMM.doNotUseTristateCorrection();
-        return Collections.singletonList(avxPairHMM);
+
+        for (VectorLoglessPairHMM.Implementation imp : VectorLoglessPairHMM.Implementation.values()) {
+            try {
+                final N2MemoryPairHMM avxPairHMM = new VectorLoglessPairHMM(imp, args);
+                avxPairHMM.doNotUseTristateCorrection();
```

Comment:
I am not sure why it's there. The line was present in the previous version of the test; I left it because I wasn't sure what it did. Looking at VectorLoglessPairHMM code, it doesn't look like it's needed. I will only comment it out in case I am wrong. (Keep in mind I'm still new to this codebase and project :-) )

The rest of your comments are addressed by a commit I am about to push.